### PR TITLE
Sys: Adding defgroup to rpl.h

### PR DESCRIPTION
**Pull request by kushalsingh007**
- Adding @defgroup and brief section to rpl.h
- Part of fix for #2656

### CODE REVIEW

**Comment by kushalsingh007** [comment]
Since I wasn't exactly sure what the exact line was I thought opening a PR and having your feedback would be a better idea.
@gebart :Can you list out the one's that you were referring to in #2656  i.e the one's for which  rpl headers are completely missing Doxygen comments,and the one's which have Doxygen commands, but inside normal comment blocks. I plan to make all those minor changes in this PR.

File: sys/net/include/rpl.h [hunk 1]
```diff
@@ -8,7 +8,11 @@
  */
 
 /**
+ * @defgroup    rpl RPL Header
```
Comment (authmillenon) [inline sys/net/include/rpl.h]: According to #2422 this should be `net_rpl`. I also would remove the "Header". The other protocol implementations don't use it either: http://riot-os.org/api/group__net.html

File: sys/net/include/rpl.h [hunk 2]
```diff
@@ -8,7 +8,11 @@
...
+ * @brief       Provides core functionality for RPL 
+ * 
  * @ingroup     rpl
```
Comment (authmillenon) [inline sys/net/include/rpl.h]: This should be `net`

**Comment by gebart** [comment]
@kushalsingh007 
 - `sys/net/include/rpl/rpl_dodag.h` is missing everything
 - `sys/net/include/rpl/rpl_of_manager.h` has broken `@file` documentation
 - `sys/net/include/rpl/rpl_structs.h` is missing documentation for the data structures
 - `sys/net/include/rpl/rpl_config.h` is missing documentation for the macros and data types

**Comment by cgundogan** [comment]
@kushalsingh007 are you still working on this?

**Comment by kushalsingh007** [comment]
@cgundogan : I do have some of these in my local repo , however for those which need entire documentation to be written, I am afraid I might not be the best person to write documentation for those one's. As I do not have the idea about those one's in the big picture.

Still, I think I'll try to put here whatever I have done soon ( once my exams end in 2 days), Thanks for the reminder :)

**Comment by cgundogan** [comment]
I close this PR, because it references the former `rpl` implementation

**Pull request closed without merging**